src/recovery/cubic.rs

```diff
@@ -213,13 +213,12 @@ fn on_packet_acked(
     // a few packets (less than RESTORE_COUNT_THRESHOLD), it's considered
     // as spurious and restore to the previous state.
     if r.congestion_recovery_start_time.is_some() {
-        if let Some(prior) = &r.cubic_state.prior {
-            let new_lost = r.lost_count - prior.lost_count;
+        let new_lost = r.lost_count - r.cubic_state.prior.lost_count;
 
-            if r.congestion_window < prior.congestion_window &&
-                new_lost < RESTORE_COUNT_THRESHOLD
-            {
-                rollback(r);
+        if new_lost < RESTORE_COUNT_THRESHOLD {
+            let did_rollback = maybe_rollback(r);
```

Comment:
What's the reason behind this change? Here we can safely assume checkpoint() is already called (it's already in recovery episode), calling rollback() doesn't need additional check.

Reply:
See the second commit's message. This is just to make the rollback avoidance if prior cwnd is smaller more generic, so each called to `rollback()` doesn't need to check this (i.e. we need this for adaptive reordering too).

Reply:
So basically this moves the ``if r.congestion_window < prior.congestion_window` into `rollback()` directly, but here we need to check if the rollback actually happened (or at least that's what the previous code did), hence `maybe_rollback()`.

Reply:
I see. If so, why not adding `bool` type to `rollback()` and merge with `maybe_rollback()`?

Reply:
No real reason. I just thought this might be slightly smaller, but your way is fine to me too. Updated commit to do that.